optaplanner-core/pom.xml

```diff
@@ -93,6 +93,12 @@
       <groupId>com.thoughtworks.xstream</groupId>
       <artifactId>xstream</artifactId>
     </dependency>
+    <!-- For (deprecated) JavaScript support. -->
+    <dependency>
+      <groupId>org.mozilla</groupId>
+      <artifactId>rhino</artifactId>
+      <scope>provided</scope>
```

Comment:
Provided is incorrect here. That indicates that it will be there at runtime through the application container.
I 'd argue it should be scope runtime (no rhino imports) with optional>true<